quinn-h3/src/frame.rs

```diff
@@ -0,0 +1,272 @@
+use std::fmt;
+use std::mem::size_of;
+
+use bytes::{Buf, BufMut, Bytes};
+
+use quinn_proto::coding::{BufExt, BufMutExt, Codec, UnexpectedEnd};
+use quinn_proto::varint;
+
+#[derive(Debug, PartialEq)]
+pub enum Error {
+    Malformed,
+    UnsupportedFrame,
+    InvalidFrameValue,
+}
+
+#[derive(Debug, PartialEq)]
+pub enum HttpFrame {
+    Data(DataFrame),
+    Settings(SettingsFrame),
+}
+
+impl HttpFrame {
+    pub fn encode<T: BufMut>(&self, buf: &mut T) {
+        match self {
+            HttpFrame::Data(f) => f.encode(buf),
+            HttpFrame::Settings(f) => f.encode(buf),
+        }
+    }
+
+    pub fn decode<T: Buf>(buf: &mut T) -> Result<Self, Error> {
+        let len = varint::read(buf).ok_or(Error::Malformed)?;
```

Comment:
`BufExt` defines `fn get_var(&mut self) -> coding::Result<u64>`, which would be more concise and preclude needing to publicize/import `proto::varint`.